pilot/cmd/pilot-agent/status/server.go

```diff
@@ -706,25 +707,11 @@ func (s *Server) handleAppProbeHTTPGet(w http.ResponseWriter, req *http.Request,
 	}
 
 	// Forward incoming headers to the application.
+	appReq.Host = req.Host
 	for name, values := range req.Header {
-		newValues := make([]string, len(values))
-		copy(newValues, values)
-		appReq.Header[name] = newValues
-	}
-
-	// If there are custom HTTPHeaders, it will override the forwarding header
-	if headers := prober.HTTPGet.HTTPHeaders; len(headers) != 0 {
```

Comment:
Do we need to comment on the `HTTPHeaders` as deprecated?

Reply:
Its not really "deprecated"; kubelet uses it. We don't need to use it since kubelet already did.